packages/keycloak-user-management/src/components/CreateEditUser/index.tsx

```diff
@@ -36,112 +38,112 @@ export interface RouteParams {
 /** props for editing a user view */
 export interface EditUserProps {
   keycloakUser: KeycloakUser | null;
-  serviceClass: typeof KeycloakService;
-  opensrpServiceClass: typeof OpenSRPService;
   keycloakBaseURL: string;
   opensrpBaseURL: string;
-  fetchKeycloakUsersCreator: typeof fetchKeycloakUsers;
   extraData: Dictionary;
 }
 
 /** type intersection for all types that pertain to the props */
 export type CreateEditPropTypes = EditUserProps & RouteComponentProps<RouteParams>;
 
-/** default props for editing user component */
-export const defaultEditUserProps: EditUserProps = {
-  keycloakUser: null,
-  serviceClass: KeycloakService,
-  opensrpServiceClass: OpenSRPService,
-  keycloakBaseURL: '',
-  opensrpBaseURL: '',
-  fetchKeycloakUsersCreator: fetchKeycloakUsers,
-  extraData: {},
-};
-
 /**
  *
  * @param props - CreateEditUser component props
  */
-
 const CreateEditUser: React.FC<CreateEditPropTypes> = (props: CreateEditPropTypes) => {
-  const [isLoading, setIsLoading] = React.useState<boolean>(false);
-  const [practitioner, setPractitioner] = React.useState<Practitioner | undefined>(undefined);
-  const {
-    keycloakUser,
-    keycloakBaseURL,
-    opensrpBaseURL,
-    serviceClass,
-    opensrpServiceClass,
-    fetchKeycloakUsersCreator,
-    extraData,
-  } = props;
+  const [userGroups, setUserGroups] = useState<UserGroup[]>([]);
+  const [initialValues, setInitialValues] = useState<FormFields>({
+    firstName: '',
+    id: '',
+    lastName: '',
+    username: '',
+    active: false,
+    userGroup: undefined,
+    practitioner: undefined,
+  });
+
+  const { keycloakUser, keycloakBaseURL, opensrpBaseURL, extraData } = props;
   const userId = props.match.params[ROUTE_PARAM_USER_ID];
-  const initialValues = keycloakUser ? keycloakUser : defaultInitialValues;
+
+  useEffect(() => {
+    if (keycloakUser && !initialValues.username) {
```

Comment:
Just curious why we need the condition `!initialValues.username`?

Reply:
if there is some keycloack user passed from the store but there is not set in the initial form values then simply set it